eessi/testsuite/common_config.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+perflog_format = '|'.join([
+    '%(check_job_completion_time)s',
+    '%(osuser)s',
+    '%(version)s',
+    '%(check_unique_name)s',
+    '%(check_info)s',
+    '%(check_system)s',
+    '%(check_partition)s',
+    '%(check_environ)s',
+    '%(check_exclusive_access)s',
+    '%(check_num_tasks)s',
+    '%(check_num_cpus_per_task)s',
+    '%(check_num_tasks_per_node)s',
+    '%(check_num_gpus_per_node)s',
+    '%(check_use_multithreading)s',
+    '%(check_modules)s',
+    '%(check_jobid)s',
+    '%(check_perfvalues)s',
+])
+
+format_perfvars = '|'.join([
+    '%(check_perf_var)s',
+    '%(check_perf_value)s',
+    '%(check_perf_lower_thres)s',
+    '%(check_perf_upper_thres)s',
+    '%(check_perf_unit)s',
+    ''  # final delimiter required
+])
+
+
+def common_logging_config(prefix=None):
+    """
+    return default logging configuration as a list: stdout, file log, perflog
+    :param prefix: file log prefix
+    """
+    prefix = os.getenv('RFM_PREFIX', prefix if prefix else '.')
+    logdir = os.path.join(prefix, 'logs')
+    os.makedirs(logdir, exist_ok=True)
+
+    return [{
+        'level': 'debug',
+        'handlers': [
+            {
+                'type': 'stream',
+                'name': 'stdout',
+                'level': 'info',
+                'format': '%(message)s',
+            },
+            {
+                'type': 'file',
+                'name': os.path.join(logdir, 'reframe.log'),
+                'level': 'debug',
+                'format': '[%(asctime)s] %(levelname)s: %(check_info)s: %(message)s',
+                'append': True,
+                'timestamp': "%Y%m%d_%H%M%S",  # add a timestamp to the filename (reframe_<timestamp>.log)
+            },
+        ],
+        'handlers_perflog': [
+            {
+                'type': 'filelog',
+                'prefix': '%(check_system)s/%(check_partition)s',
```

Comment:
I'm wondering if this should also be prefixed with `logdir`...

Currently, if I set the system's `prefix`, but don't pass that as argument to `common_logging_config`, the `perflogs` will end up in the system's `prefix`, whereas the `logs` will be in the `logdir` (see what happened for Snellius above). That feels strange. It would feel more natural if this `common_logging_config` would _always_ keep `logs` and `perflogs` together, regardless of whether an argument is passed or not.

Reply:
i agree it would be better to always keep everything together, but with your suggestion the `output` directory would still be in a different location from `logs` and `perflogs`, so this is not a solution.

the real solution would be to fix prefix issue for the `perflogs`, see https://github.com/reframe-hpc/reframe/issues/2990
Vasileios seemed open for supporting it.

in the mean time i would recommend to always set `RFM_PREFIX`, not set `--prefix`, and not set a "system `prefix`" in the config file.
if you really want to set a "system `prefix`", just make sure it's passed to `common_logging_config` and all will be fine.

Reply:
True. I just meant that having `output` in a different prefix than `logs` and `perflogs` imho is 'ok-ish' (though not ideal). Well, let's not worry about it too much: it seems Vasileous is indeed open to changing this behaviour, so we shouldn't worry too much on our side.

> in the mean time i would recommend to always set RFM_PREFIX, not set --prefix, and not set a "system prefix" in the config file. if you really want to set a "system prefix", just make sure it's passed to common_logging_config and all will be fine.

True. Maybe my practice of setting the 'system prefix' in the config file isn't a great one... Then again, the advantage is that if I give this to users, and I later need to help them debug issues, at least I know where to find their logs! :D

Ok, for now, let's keep it as is. Who knows, in the future, I might remove setting the system prefix from the config file...